### DISCUSSION TITLE
How do you retrieve a form object to add to a task?

### DISCUSSION THREAD

**Discussion by fjhaag** [sadieblue/agilityblue.workspace-scripts#2] [Q&A] [answered]
I am working on a script to add QC forms when a task is placed into QC, when I call Add-FormsToTask I receive the following error message:

[Uncaught Exception] Cannot bind parameter 'Forms'. Cannot convert the AgilityBlue.ViewModels.Workspace.FormViewModel" value of type "AgilityBlue.ViewModels.Workspace.FormViewModel" to type "AgilityBlue.ViewModels.Workspace.TaskFormViewModel".

How should I retrieve the form object so that is of the correct type?

Here is the code that I have:
```powershell
$forms = $task.Forms.FormId
$formsToAdd = @()

foreach ($form in $forms){
    Write-Output "Looking through forms."
    if ($formTable.Values -contains $form) {
        Write-Output "QC form exists."
    } elseif ($formTable.Keys -contains $form) {
        # formTable is a hash table with the key being the form id of the existing form and the value being the associated qc form id
        $formsToAdd += Get-Form $formTable[$form]
        Write-Output "QC form to add found."
    } else {
        Write-Output "No QC form for form $($form)."
    }
}

if ($formsToAdd.Count -gt 0) {
        Add-FormsToTask -TaskId $task.TaskId -Forms $formsToAdd
        Write-Output "Added forms $($formsToAdd.FormId) to Task $($task.TaskId)."
    } else {
        Write-Output "No QC forms to Add."
}
```

Any help is appreciated. Thank you,

**Comment by bmward80** [comment, marked as answer]
HI, @fjhaag,

There's a couple of script examples in the repository that create tasks forms from forms:

- [Create a QC task when a processing task is completed.ps1](https://github.com/sadieblue/agilityblue.workspace-scripts/blob/master/Create%20a%20QC%20task%20when%20a%20processing%20task%20is%20completed.ps1)
- [Create a set of projects from templates when a new matter is created.ps1](https://github.com/sadieblue/agilityblue.workspace-scripts/blob/master/Create%20a%20set%20of%20projects%20from%20templates%20when%20a%20new%20matter%20is%20created.ps1)

It's not a direct example of the Add-FormsToTask cmdlet like you're asking, but the concepts of building up the task forms are there that might help give you some guidance on the form structures needed.

Let me know if that helps or if you get stuck and we'll see if we can put together a direct example.

Brandon

**Reply by fjhaag** [reply]
With the Create a QC task example I was able to get it figured out. Thank you!

**Reply by bmward80** [reply]
Great, happy to help! 👍
